server/admin_management/api/app/routers/router_guards.py

```diff
@@ -124,103 +140,66 @@ async def get_target_user_from_id(
+                "user_guid": rbody["user_guid"],
+            }
         )
-        if user is not None:
-            found_target_user = TargetUser.model_validate(user)
-            # old data might not have user_guid.
-            if found_target_user.user_guid is None:
-                found_target_user.user_guid = rb_user_guid
-            return found_target_user
-        # user not found, case of new user.
-        else:
-            target_new_user = TargetUser.model_validate({
-                "user_name": rb_user_name,
-                "user_type_code": rb_user_type_code,
-                "user_guid": rb_user_guid
-            })
-            return target_new_user
+        return target_new_user
 
 
-async def target_user_bceid_search(
+async def get_verified_target_user(
     requester: Requester = Depends(get_current_requester),
-    _target_user: TargetUser = Depends(get_target_user_from_id)
+    _target_user: TargetUser = Depends(get_target_user_from_id),
 ) -> TargetUser:
     """
-    BCeID user search for target_user.
-    :param target_user "Depends(get_target_user_from_id)": the initial
-        dependency with "get_target_user_from_id" has initial target_user
-        parsed from the request. It's business_guid will be updated to the
-        target_user after user is searched through IDIM Proxy.
+    Validate the target user by calling IDIM web service, and update business Guid for the found BCeID user
     """
     target_user = copy.deepcopy(_target_user)
-    if (
-        target_user.user_type_code == UserType.BCEID and
-        target_user.business_guid is None
-    ):
-        user_guid = target_user.user_guid  # will be used by search using user_guid
-        LOGGER.debug(
-            f"Searching business guid for target_user: {target_user}, with requester {requester}"
-        )
-        idim_proxy_api = IdimProxyService(requester)
-        search_result = idim_proxy_api.search_business_bceid(
-            IdimProxyBceidSearchParam(**{
-                "searchUserBy": IdimSearchUserParamType.USER_GUID,
-                "searchValue": user_guid
-            })
-        )
-        if search_result.get("found"):
-            business_guid = search_result.get("businessGuid")
-            LOGGER.debug(
-                f"business_guid result: {business_guid} is updated to target_user."
-            )
-            target_user.business_guid = business_guid
-        else:
-            LOGGER.debug(
-                f"IDIM look up user guid:`{user_guid}` returns no result. " +
-                "business_guid will NOT be updated to target_user."
-            )
+    idim_validator_service = IdimValidator(requester, target_user)
+    search_result = idim_validator_service.verify_user_exist()
 
+    # add business_guid for BCeID user
+    if search_result.get("found") and search_result.get("businessGuid"):
+        target_user.business_guid = search_result.get("businessGuid")
     return target_user
```

Comment:
Update the bceid search to a new method "get_verified_target_user", and we'll call the idim web service to find the user. If returns businessGuid, will add to the target user